### PR TITLE
refa[DST-598]: improve range slider

### PR DESCRIPTION
**Pull request by aromko**
# Description

The slider is completely refactored. I oriented myelf on https://react-spectrum.adobe.com/react-aria/Slider.html.

- the stories are updated, also added new stories
- add a visibile “indicator” for the selected track
- added Fieldbase so Helptext can now be used
- updated/added tests
- rewrite slider documentation according to our component pages

- [x] This change requires a UI-Kit update - inform Alex Tirado (@tirado-rx), Benji @benjirsvx 

# What should be tested?

- Storybook
- Documentation

# Reviewers:

@marigold-ui/developer
@tirado-rx or @benjirsvx

### CODE REVIEW

File: themes/theme-core/src/components/Slider.styles.ts [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import { ThemeComponent, cva } from '@marigold/system';
 
 export const Slider: ThemeComponent<'Slider'> = {
-  container: cva(),
+  container: cva([
```
[the remark this answers is not in the document]
Comment (aromko) [reply themes/theme-core/src/components/Slider.styles.ts:4]: It's the workaround to make it work with Fieldgroup and their label.

[the remark this answers is not in the document]
Comment (aromko) [reply themes/theme-core/src/components/Slider.styles.ts:4]: @sebald ping

File: docs/content/components/form/slider/slider.mdx [hunk 2]
```diff
@@ -1,66 +1,213 @@
...
 
+<AppearanceDemo component={title} />
+
 <AppearanceTable component={title} />
 
-## Props
+## Usage
 
-<PropsTable component={title} />
+Sliders allow users to view and select a value (or range) along a track. They’re ideal for adjusting settings such as volume and brightness.
```
Comment (sebald) [inline docs/content/components/form/slider/slider.mdx:40]:
```suggestion
Sliders allow users to view and select a value (or range) along a track. They’re ideal for adjusting settings such as volume and brightness.

```

File: docs/content/components/form/slider/slider.mdx [hunk 3]
```diff
@@ -1,66 +1,213 @@
...
+A slider’s `step` size defines the increments at which the slider can move, allowing users to make finer or coarser adjustments based on the context—such as adjusting volume in single steps versus increments of ten. Setting an appropriate step size enhances usability, especially for precise inputs.
```
Comment (sebald) [inline docs/content/components/form/slider/slider.mdx:41]:
```suggestion
A slider’s `step` size defines the increments at which the slider can move, allowing users to make finer or coarser adjustments based on the context — such as adjusting volume in single steps versus increments of ten. Setting an appropriate step size enhances usability, especially for precise inputs. The `step` default increment is 1.
```

Added whitespace to `—` (AI is forgetting that all the time ;D) and moved the default sentence to the paragraph.

File: docs/content/components/form/slider/slider.mdx [hunk 4]
```diff
@@ -1,66 +1,213 @@
...
 
-## Examples
+The `step` default increment is <strong>1</strong>.
```
Comment (sebald) [inline docs/content/components/form/slider/slider.mdx:43]:
```suggestion
```

**Review by sebald — CHANGES_REQUESTED** [review]

**Pull request merged**